File: solutions/go/acronym/1/acronym.go
// Package acronym get the acronym given a string
package acronym

import ("unicode"
        "strings")

// Abbreviate get the acronym 
func Abbreviate(s string) string {
    var sb strings.Builder
	for i, r := range s {
        if unicode.IsLetter(r) {
            if i == 0 || IsSeparator(rune(s[i-1])){
                sb.WriteRune(unicode.ToUpper(r))
            }
        }
    }

    return sb.String()
}

func IsSeparator(r rune) bool {
    switch r {
        case '-':
        return true
        case ' ':
        return true
        case '_':
        return true
        default:
        return false
    }
}

File: solutions/csharp/acronym/1/Acronym.cs
public static class Acronym
{
    public static string Abbreviate(string phrase)
    {
        var words = phrase.Split([' ', '-', '_'], StringSplitOptions.RemoveEmptyEntries);
        var acronym = new System.Text.StringBuilder();
        foreach (var word in words)
        {
            if (char.IsLetter(word[0]))
                acronym.Append(char.ToUpper(word[0]));
        }
        return acronym.ToString();
    }
}